### PR TITLE
Add logical metadata file validation, fixes #98

### LINKED ISSUE #98: Problem: migration packages will now contain a logical metadata file
**Describe the bug**

The migration packages will include an extra metadata file going forward (the logical metadata extracted from Preservica). SFA would like there to be a check to confirm that the file has been included in the package, that it is valid, and then move it to the metadata directory. This affects the **BornDigitalAIP** and **DigitizedAIP** transfer types.

**Expected behavior**

Some activity (could re-use the _Validate SIP structure_ activity) will look for a file located at `package-name/data/additional/package-name-premis.xml`. The activity will confirm that the file is present and that the name of the file preceding `-premis.xml` matches the name of the package. 

Another activity (perhaps _Validate SIP metadata_ or _Validate PREMIS_) will validate the above listed file. It should be valid PREMIS. The output for the activity should indicate that this PREMIS file has been validated.

Finally, the file should be moved to the metadata directory during the _Restructure SIP_ activity.

I will provide sample packages once the structure is confirmed by SFA.

**Additional context**

See #97 for more information about the new package structure.

### PR DESCRIPTION
**Pull request by djjuhasz**
- Add the logical metadata file to the expected SIP structure for AIP type SIPs
- Validate the logical metadata file against the PREMIS v3 XSD
- Move the logical metadata file to the "metadata/" directory when transforming the SIP for delivery to preservation

### CODE REVIEW

File: internal/activities/validate_premis_test.go [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+					fs.WithFile("premis.xml", premisXML),
+				).Join("premis.xml"),
+			},
+			want: activities.ValidatePREMISResult{
+				Failures: []string{"premis.xml does not match expected metadata requirements"},
+			},
+		},
+		{
+			name:      "Returns a file not found failure",
+			validator: newFakeValidator().WithErr(errors.New("file not found")),
```
Comment (mcantelon) [inline internal/activities/validate_premis_test.go:97]: What are the pros and cons of having a mock object return a file not found error versus testing with a non-existent file?

File: internal/activities/validate_premis_test.go [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+				if err == nil {
+					t.Errorf("error is nil, expecting: %q", tt.wantErr)
+				} else {
+					assert.ErrorContains(t, err, tt.wantErr)
+				}
+
+				return
+			}
+
+			t.Log(logbuf.String()) // Echo log for debugging.
```
Comment (mcantelon) [inline internal/activities/validate_premis_test.go:146]: Slick!

File: internal/activities/validate_structure.go [hunk 3]
```diff
@@ -75,14 +82,24 @@ func (a *ValidateStructure) Execute(
...
+	if params.SIP.Type == enums.SIPTypeDigitizedSIP || params.SIP.Type == enums.SIPTypeDigitizedAIP && hasContentDir {
 		entries, err := os.ReadDir(params.SIP.ContentPath)
 		if err != nil {
 			return nil, fmt.Errorf("ValidateStructure: check for unexpected dossiers: %v", err)
 		}
 
-		if len(entries) > 1 {
+		dirs := 0
+		for _, e := range entries {
+			if e.IsDir() {
```
Comment (mcantelon) [inline internal/activities/validate_structure.go:94]: Good catch!

File: internal/premis/xsdv3.go [hunk 4]
```diff
@@ -0,0 +1,6 @@
+package premis
+
+import _ "embed"
+
+//go:embed premis-v3-0.xsd
```
Comment (mcantelon) [inline internal/premis/xsdv3.go:5]: Very handy feature, this embed magic!

File: internal/sip/sip.go [hunk 5]
```diff
@@ -114,3 +118,11 @@ func (s SIP) bornDigitalSIP() SIP {
 func (s SIP) Name() string {
 	return filepath.Base(s.Path)
 }
+
+func (s SIP) IsAIP() bool {
```
Comment (mcantelon) [inline internal/sip/sip.go:122]: Good thinking moving these here!

**Review by mcantelon — APPROVED** [review]
This seems great! I had one question (for my own curiosity), but see any issues... the embedding of the PREMIS XSD is slick.

File: internal/activities/validate_premis_test.go (hunk 1 above)
Comment (djjuhasz) [reply internal/activities/validate_premis_test.go:97]: @mcantelon I think the main pro of using a mock validator is that the tests are not dependent on xmllint being installed in the test environment. Unfortunately I negated that advantage by using the real "xmllint" validator in the first test, and sure enough that test failed in CI until I added xmllint to the CI runner. :thinking: The mock validator is also faster (though I'm sure the difference isn't noticeable in these small test cases), and allows more control over the return values. 

I think the main con of using a mock validator is if there is some integration issue between the `ValidatePREMIS` activity and `xmlvalidate.XMLLintValidator` then it won't be caught by these unit tests.  Ideally integration issues would be caught by integration tests, but we don't have any integration tests for this project. :(

File: internal/activities/validate_premis_test.go (hunk 2 above)
Comment (djjuhasz) [reply internal/activities/validate_premis_test.go:146]: Yeah! It took me a while to figure out how to capture and test the log messages. :sweat_smile:

File: internal/activities/validate_premis_test.go (hunk 1 above)
Comment (mcantelon) [reply internal/activities/validate_premis_test.go:97]: Good points! Yeah, integration tests definitely seem like they'd be handy (but also potentially a lot of work to create/maintain at this stage).

File: internal/sip/sip.go (hunk 5 above)
Comment (djjuhasz) [reply internal/sip/sip.go:122]: Yeah, it saves some repeat code. :)

File: internal/activities/validate_premis_test.go (hunk 1 above)
Comment (djjuhasz) [reply internal/activities/validate_premis_test.go:97]: I've decided to remove the xmllint dependency in the first test here.  The deciding factor for me is that the tests should not fail if a developer runs them locally without xmllint installed.

**Pull request merged**